client/util.go

```diff
@@ -120,3 +110,28 @@ func depadPathname(paddedPathname []byte) (string, error) {
 
 	return string(buf.Next(int(origLen))), nil
 }
+
+// relPathStrict returns a relative path for `targpath` from `basepath`.  Unlike
+// the `filepath.Rel` function, this function returns an error if `targpath` is
+// not within `basepath`.
+func relPathStrict(basepath, targpath string) (string, error) {
+	absTargpath, err := filepath.Abs(targpath)
+	if err != nil {
+		return "", err
+	}
+
+	absBasepath, err := filepath.Abs(basepath)
+	if err != nil {
+		return "", err
+	}
+
+	if !strings.HasPrefix(absTargpath, absBasepath) {
```

Comment:
Do we need to make sure `absBasepath` ends in a `/`, so we don't see false prefixes as in `strings.HasPrefix("somedirectory/x", "somedir")`?